src/config/applyOverrides.js

```diff
@@ -0,0 +1,49 @@
+const debug = require('debug')('npm-package-json-lint:applyOverrides');
+const path = require('path');
+const globby = require('globby');
+
+/**
+ * Applies values from the 'overrides' field in a configuration file.
+ * @param {string} cwd       The current working directory.
+ * @param {Object} filePath  The file path of the file being linted.
```

Comment:
@tclindner is this working as intended for you? I'm seeing the path of the config file as `filePath` instead of file being linted. Therefore `filePath === globbedFilePath` is never truthy and overrides don't work.

Note how path gets pulled from config object here before passed to `applyOverrides`:

https://github.com/tclindner/npm-package-json-lint/pull/123/files#diff-4c88ce73413b52beff7db767b94bd2f4R11

Reply:
Hey @simison - it seems to be working for my basic setup. What does your setup look like? We can try to get a patch in.

Reply:
Heh, well you just got me to doubt myself and if I was testing thoroughly enough! 😅 

Here's the config I'm working on https://github.com/Automattic/wp-calypso/pull/36534

No worries if you don't have time, I can also dig deeper into this.

Reply:
Thank you, @simison! I think I have a fix ready. Please give v4.0.1 a shot and let me know if it resolves your issue. Thanks again for reporting it 👍

Reply:
Yasss, that worked fantastically! Thank you!